judo-runtime-core-dao-rdbms/src/main/java/hu/blackbelt/judo/runtime/core/dao/rdbms/RdbmsDAOImpl.java

```diff
@@ -736,22 +736,24 @@ protected Collection<Payload> readRangeOf(final EReference reference, final Payl
                     .map(p -> markSelected.apply(p, currentReferences))
                     .collect(Collectors.toList());
         } else {
-            final Payload temporaryInstance;
-            if (instanceId != null) {
-                temporaryInstance = update(reference.getEContainingClass(), payload, QueryCustomizer.<ID>builder()
-                        .mask(Collections.emptyMap())
-                        .build(), false);
-            } else if (payload != null) {
-                temporaryInstance = create(reference.getEContainingClass(), payload, QueryCustomizer.<ID>builder()
-                        .mask(Collections.emptyMap())
-                        .build(), false);
-                instanceId = temporaryInstance.getAs(identifierProvider.getType(), identifierProvider.getName());
-            } else {
-                throw new IllegalArgumentException("Missing input to get range");
-            }
+            if (stateful) {
+                final Payload temporaryInstance;
+                if (instanceId != null) {
+                    temporaryInstance = update(reference.getEContainingClass(), payload, QueryCustomizer.<ID>builder()
+                            .mask(Collections.emptyMap())
+                            .build(), false);
+                } else if (payload != null) {
+                    temporaryInstance = create(reference.getEContainingClass(), payload, QueryCustomizer.<ID>builder()
+                            .mask(Collections.emptyMap())
+                            .build(), false);
+                    instanceId = temporaryInstance.getAs(identifierProvider.getType(), identifierProvider.getName());
+                } else {
+                    throw new IllegalArgumentException("Missing input to get range");
+                }
 
-            if (log.isDebugEnabled()) {
-                log.debug("Saved temporary instance {} with ID: {}", AsmUtils.getClassifierFQName(reference.getEContainingClass()), temporaryInstance.get(identifierProvider.getName()));
+                if (log.isDebugEnabled()) {
+                    log.debug("Saved temporary instance {} with ID: {}", AsmUtils.getClassifierFQName(reference.getEContainingClass()), temporaryInstance.get(identifierProvider.getName()));
+                }
```

Comment:
I statless mode when instanceId is null have to throw an IllegalArgumentException.

Reply:
+1

Reply:
[214ef2f](https://github.com/BlackBeltTechnology/judo-runtime-core/pull/149/commits/214ef2f39e2556c6da0710638c1a33cc551686e4)